File: 백준/Silver/1931. 회의실 배정/회의실 배정.swift
import Foundation

let n = Int(readLine()!)!
var times:[[Int]] = []

for _ in 0..<n {
    let input = readLine()!.split(separator: " ").map{Int(String($0))!}
    times.append(input)
}
times.sort { a,b in
    if (a[1]==b[1]) { return a[0] <= b[0] }
    else { return a[1] <= b[1] }
}

var end_time = times[0][1]
var answer = 1
for i in 1..<n {
    if(times[i][0] >= end_time) { 
        answer += 1
        end_time = times[i][1]
    }
}
print(answer)

File: 백준/Gold/1931. 회의실 배정/회의실 배정.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        
        int N = Integer.parseInt(br.readLine());

        int[][] times = new int[N][2];
        for(int i=0; i<N; i++) {
            st = new StringTokenizer(br.readLine());
            times[i][0] = Integer.parseInt(st.nextToken());
            times[i][1] = Integer.parseInt(st.nextToken());
        }
        Arrays.sort(times, new Comparator<int[]>(){
            @Override
            public int compare(int[] o1, int[] o2) {
                if(o1[1] == o2[1]) return o1[0]-o2[0];
                else return o1[1]-o2[1];
            }
        });

        int end_time = times[0][1];
        int answer = 1;
        for(int i=1; i<N; i++) {
            if(times[i][0] >= end_time) {
                answer++;
                end_time = times[i][1];
            }
        }
        System.out.println(answer);
    }
}